src/Question/question.controller.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  ApiBasicAuth,
+  ApiForbiddenResponse,
+  ApiOkResponse,
+  ApiQuery,
+  ApiTags,
+} from "@nestjs/swagger";
+import {
+  CacheInterceptor,
+  ClassSerializerInterceptor,
+  Controller,
+  Get,
+  UseInterceptors,
+  Query,
+} from "@nestjs/common";
+
+import { QuestionService } from "src/adapters/default/vidyadanQuestion.adapter";
+
+@ApiTags("Question")
+@Controller("question")
+export class QuestionController {
+  constructor(private service: QuestionService) {}
+
+  @Get("vidyadan/getall")
```

Comment:
The routes need to have the adapter name as a path variable so `questions/:adapter/search` which will internally call the search method from the `:adapter` eg: from the diksha adapter.